src/containers/App/index.js

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from 'react';
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+import Paper from 'material-ui/Paper';
+
+import Header from '../../components/Header';
+import TodoInput from '../../components/TodoInput';
+import TodoList from '../../containers/TodoList';
+
+import './styles.css';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      todoInput: {
+        text: '',
+      },
+      lastId: 0,
+      filter: 'none',
+      todos: []
+    }
+
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.toggleComplete = this.toggleComplete.bind(this);
+    this.deleteCompleted = this.deleteCompleted.bind(this);
+    this.deleteTodo = this.deleteTodo.bind(this);
+    this.handleFilterChange = this.handleFilterChange.bind(this);
+    this.submitTodo = this.submitTodo.bind(this);
+  }
+
+  handleInputChange(input, e) {
+    const updateInputState = {};
+    updateInputState[input] = e.target.value;
+    this.setState({ todoInput: { ...this.state.todoInput, ...updateInputState } });
+  }
+
+  submitTodo(todo, e) {
+    e.preventDefault();
+    const { todos } = this.state;
+    let { lastId } = this.state;
+
+    this.setState({ todos: [...todos, { id: lastId, ...todo, complete: false }] });
+    this.setState({ todoInput: { text: '' } });
+    this.setState({ lastId: ++lastId })
+  }
```

Comment:
Is there  a reason for using multiple setState invocations here? You could write all of your state updates as a single setState call.

eg:

```js
this.setState({
    todos: [ ... ],
    todoInput: { ... },
    lastId: ++...
})
```